snapd.go

```diff
@@ -702,6 +706,53 @@ func setMaxProcs(maxProcs int) {
 	}
 }
 
+// UnmarshalJSON unmarshals valid json into a Config.  An example Config can be found
+// at github.com/intelsdi-x/snap/blob/master/examples/configs/snap-config-sample.json
+func (c *Config) UnmarshalJSON(data []byte) error {
+	// construct a map of strings to json.RawMessages (to defer the parsing of individual
+	// fields from the unmarshalled interface until later), then unmarshal the input
+	// byte array into that map
+	t := make(map[string]json.RawMessage)
+	if err := json.Unmarshal(data, &t); err != nil {
+		return err
+	}
+	// loop through the individual map elements, parse each in turn, and set
+	// the appropriate field in this configuration
+	for k, v := range t {
+		switch k {
+		case "log_level":
+			if err := json.Unmarshal(v, &(c.LogLevel)); err != nil {
+				return err
+			}
+		case "gomaxprocs":
+			if err := json.Unmarshal(v, &(c.GoMaxProcs)); err != nil {
+				return err
+			}
+		case "logpath":
```

Comment:
fixed, good catch though...I guess that's what comes of typing these things in instead of copying and pasting :)